### PR TITLE
Replace sed with envsubst

### PR DESCRIPTION
**Pull request by aeneasr**

### CODE REVIEW

File: scripts/update-repository-templates.sh [hunk 1]
```diff
@@ -1,62 +0,0 @@
-#!/bin/bash
```
Comment (aeneasr) [inline scripts/update-repository-templates.sh:1]: @vinckr this script wasn't actually used anywhere any more.

File: scripts/sync.sh [hunk 2]
```diff
@@ -31,12 +31,9 @@ function sync {
...
     cp -R "templates/repository/$type/.github" "$workdir/"
 
-    # Replace placeholders
-    if [[ "$OSTYPE" == "darwin"* ]]; then
-      sed -e "s|{{Project}}|$humanName|g" `find "$workdir/.github" -type f -print` "$workdir/CONTRIBUTING.md" "$workdir/SECURITY.md"
-    else
-      sed -i -e "s|{{Project}}|$humanName|g" `find "$workdir/.github" -type f -print` "$workdir/CONTRIBUTING.md" "$workdir/SECURITY.md"
-    fi
+    for f in (find "$workdir/.github" "$workdir/CONTRIBUTING.md" "$workdir/SECURITY.md" "$workdir/CODE_OF_CONDUCT.md" "$workdir/LICENSE" -type f -print); do {
+      REPOSITORY="$project" PROJECT="$humanName" envsubst $f | sponge $f
```
Comment (aeneasr) [inline scripts/sync.sh:35]: I replaced sed with envsubst:

https://stackoverflow.com/questions/14155596/how-to-substitute-shell-variables-in-complex-text-files

File: templates/repository/common/.github/ISSUE_TEMPLATE/config.yml [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
+  - name: ORY ${PROJECT} Forum
+    url: https://github.com/${REPOSITORY}/discussions
     about: Please ask and answer questions here, show your implementations and discuss ideas.
   - name: ORY Chat
     url: https://www.ory.sh/chat
     about: Hang out with other ORY community members and ask and answer questions.
   - name: ORY Support for Business
     url: https://github.com/ory/open-source-support/blob/master/README.md
-    about: Buy professional support for ORY {{Project}}.
+    about: Buy professional support for ORY ${PROJECT}.
```
Comment (aeneasr) [inline templates/repository/common/.github/ISSUE_TEMPLATE/config.yml:11]: New envsubst syntax

File: scripts/sync.sh (hunk 2 above)
Comment (aeneasr) [reply scripts/sync.sh:35]: I also used this search `iterate over files bash find`: https://stackoverflow.com/questions/9612090/how-to-loop-through-file-names-returned-by-find

Comment (aeneasr) [reply scripts/sync.sh:35]: This means that you can use `${REPOSITORY}` in the text and `${PROJECT}` which are replaced by e.g. `ory/kratos` and `Kratos`

**Review by aeneasr — COMMENTED** [review]
@vinckr please check the comments, I added some details about the changes

File: scripts/sync.sh (hunk 2 above)
Comment (vinckr) [reply scripts/sync.sh:35]: Looks very nice! Thanks for the writeup and info around it!

**Pull request merged**